### PR TITLE
feat: use make-fetch-happen instead of node-fetch

### PR DESCRIPTION
**Pull request by ivarconr**
In respect to some network issues discovered with node-fetch, where it does not automatically do reties for common network errors I have switched out node-fetch in favor of [make-fetch-happen](https://github.com/npm/make-fetch-happen)

I also played with the [got](https://www.npmjs.com/package/got) http client. The benefit of got is that it also support http2 and seems to be even more used in the wild. Switching to got will require us to to a larger rewrite and probably some more testing and verification.

### CODE REVIEW

**Review by chriswk — APPROVED** [review]
Seems good. Is the change of streams implementation (node core -> minipass) something we'd need to worry about?

**Review by FredrikOseberg — APPROVED** [review]
LGTM

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
   "homepage": "https://github.com/Unleash/unleash-client-node",
   "dependencies": {
     "ip": "^1.1.5",
+    "make-fetch-happen": "^8.0.14",
     "murmurhash3js": "^3.0.1",
     "node-fetch": "^2.6.1"
```
Comment (SimenB) [inline package.json:34]: is this still used?

File: test/unleash.network.retries.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  t.plan(1);
+
+  let calls = 0;
+  const server = createServer((req, res) => {
+    calls ++;
+    res.writeHead(408);
+    res.end();
+  });
+
+  server.listen(async () => {
```
Comment (SimenB) [inline test/unleash.network.retries.test.js:15]:
```suggestion
  server.listen(() => {
```

File: test/unleash.network.retries.test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import test from 'ava';
+import { createServer } from 'http';
+import { Unleash } from '../lib/unleash';
+
+test.cb('should retry on error', (t) => {
+  t.plan(1);
+
+  let calls = 0;
+  const server = createServer((req, res) => {
+    calls ++;
```
Comment (SimenB) [inline test/unleash.network.retries.test.js:10]:
```suggestion
    calls++;
```

weird space, might just be me 😀

File: test/unleash.network.test.js [hunk 4]
```diff
@@ -20,6 +18,7 @@ test.cb('should emit network errors', (t) => {
   });
 
   unleash.on('error', (e) => {
+    // console.log(e);
```
Comment (SimenB) [inline test/unleash.network.test.js:21]:
```suggestion
```

**Review by SimenB — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ivarconr) [reply package.json:34]: no, will remove :+1:

File: src/request.ts [hunk 5]
```diff
@@ -44,23 +44,24 @@ export const buildHeaders = (
...
+    head['If-None-Match'] = etag;
   }
   if (contentType) {
-    head.append('Content-Type', contentType);
+    head['Content-Type'] = contentType;
   }
   if (custom) {
-    Object.keys(custom).forEach((k) => head.append(k, custom[k]));
+    // eslint-disable-next-line no-return-assign
+    Object.keys(custom).forEach((k) => (head[k] = custom[k]));
```
Comment (SimenB) [inline src/request.ts:64]:
```suggestion
    Object.assign(head, custom);
```

I think that'll work

**Pull request merged**